### ISSUE TITLE
WPF: Keep selection when calling GridView.ReloadData()

### ISSUE THREAD

**Issue by TomQv** [picoe/Eto#1360]
Sometimes data is changing and the GridView needs to be refreshed.

For this I call GridView.ReloadData(rows);

- the overload with the row-numbers is actually useless, because on the Wpf-side it all just ends at Control.Items.Refresh().

- this call is clearing the current selection. Is there a way to keep the selection and not to get a SelectionChanged-event?

btw on mac this works fine, no problems. So again a stupid wpf-thing?

**Comment by cwensley** [comment]
> the overload with the row-numbers is actually useless, because on the Wpf-side it all just ends at Control.Items.Refresh().

Correct, refreshing just a set of rows is not (easily) possible in WPF afaik, and refreshing the whole thing is much more efficient.  It only reloads the data that is visible in my testing anyway.

> this call is clearing the current selection. Is there a way to keep the selection and not to get a SelectionChanged-event?

Hm, yeah I think it should keep selection when ReloadData() is called.  Thanks for reporting the issue!

**Comment by TomQv** [comment]
Any news on this? I mean, keeping the selection with ReloadData() ?
Or is there maybe a way to "store" and "restore" the selection?
I tried this, but no success, selected items are not restored. Also don't get it, how SelectionPreserver is supposed to work
```
var rows = _list.SelectedRows;
...
_list.ReloadData();
_list.SelectedRows = rows;
```

**Comment by TomQv** [comment]
Ok, here is a workaround, that works for me. I was trying this before, but the yield messed it up, so adding ToList to really get the rows before selection is cleared in Reload did the trick:

```
var rows = _list.SelectedRows.ToList ();  // ToList() is important, because SelectedRows returns with yield..
_list.ReloadData();
_list.SelectedRows = rows;
```
